webapp/src/components/preview-movies/style/row/TopRow.tsx

```diff
@@ -1,54 +1,119 @@
-import Arrow from './Arrow';
-import Poster from '../image/Poster';
+import '../../../../../assets/scss/components/style/row/row.scss';
+import '../../../../../assets/scss/components/style/arrow/arrow.scss';
+import '../../../../../assets/scss/components/style/row/top-row.scss';
+import Arrow from '../arrow/Arrow';
+import Poster from '../poster/Poster';
 
 type Props = {
   title?: string,
   movieList: Production[],
   isDataLoading?: boolean
 };
 
+const SLIDER_TIMING: number = 8000;
+
 function TopRow({ movieList, isDataLoading }: Props) {
-  function hundleClickArrowRight() {
-    const slider = document.getElementById('top_posters');
-    slider.scrollLeft += window.innerWidth;
+  const slider = useRef<HTMLDivElement>(null);
+
+  const [currentPoster, setCurrentPoster] = useState(0);
+
+  const sliderInterval = useRef<NodeJS.Timeout>(setInterval(() => {
+    goNextPoster();
+  }, SLIDER_TIMING));
+
+  function createSliderInterval() {
+    return sliderInterval.current;
+  }
+
+  function stopSliderInterval() {
+    if (!sliderInterval) {
+      return;
+    }
+
+    return clearInterval(sliderInterval.current);
+  }
+
+  function goNextPoster() {
+    setCurrentPoster((prevPoster) => {
+      if (!slider.current) {
+        return prevPoster;
+      }
+
+      stopSliderInterval();
+
+      if (prevPoster + 1 >= movieList.length) {
+        slider.current.scrollLeft = 0;
+        return 0;
+      }
+      slider.current.scrollLeft += window.innerWidth;
+
+      createSliderInterval();
+
+      return prevPoster + 1;
+    });
   }
 
-  function hundleClickArrowLeft() {
-    const slider = document.getElementById('top_posters');
-    slider.scrollLeft -= window.innerWidth;
+  function goPreviousPoster() {
+    if (!slider.current) {
+      return;
+    }
+
+    stopSliderInterval();
+
+    setCurrentPoster(currentPoster - 1);
+    slider.current.scrollLeft -= window.innerWidth;
+
+    createSliderInterval();
   }
 
+  useEffect(() => {
+    createSliderInterval();
+
+    // Nettoyer l'intervalle lorsque le composant est démonté pour éviter les fuites de mémoire
+    return () => {
+      clearInterval(sliderInterval.current);
+    };
+  }, [movieList]);
+
   return (
-    <div className={'row'} >
+    <div className='row-top'>
       {
-        isDataLoading
-          ? <RowLoading/>
-
-          : <div className={'row_posters'} id={'top_posters'}>
-            {movieList.map((movie) => (
-              <div key={movie.title}>
-                <div className={'top_card'} style={{ width: `${window.innerWidth}px` }}>
-                  <div>
-                    <h2>{movie.title}</h2>
-                    <p>{movie.overview}</p>
-                    <Player />
-                  </div>
-                  <Poster title={movie.title} path={movie.backdrop_path} className={'top_img'}/>
+        isDataLoading ? <RowLoading/>
+          : <div ref={slider} className='row__posters row__posters--top'>
```

Comment:
saut de ligne + parentheses

```suggestion
        isDataLoading
        ? <RowLoading/>
        : (
          <div ref={slider} className='row__posters row__posters--top'>
```